EXPERIMENTS.md

```diff
@@ -97,3 +97,11 @@ We now understand this causes a bug, and we want to avoid it. Enabling this expe
 https://github.com/buildkite/agent/blob/40b8a5f3794b04bd64da6e2527857add849a35bd/internal/job/executor.go#L1980-L1993
 
 **Status:** Since the default behaviour is buggy, we will be promoting this to non-experiment soon™️.
+
+### `strict-single-hooks`
+
+We document that multiple checkout and command hooks will not run, only the first encountered will. A long-standing bug means this is not true for checkout and command hooks provided by plugins.
+
+This experiment enforces only running one checkout or command hook, even if multiple plugins define them. 
+
+**Status:** This experiment fixes a long-standing bug, so we intend to promote this to non-experiment when most pipelines and plugins have been updated.
```

Comment:
I don't think this condition will ever be met, and if it is, do we have any way of knowing?

It's also possible that for a large class of users, plugins running multiple checkout hooks is a feature they use to check out multiple repos with internal plugins, and/or multiple command hooks on plugins are a way to ensure multiple commands are run on the same agent in the same working directory without cleaning up in between.

For these reasons, I suggest ripping the band-aid off only with a major version bump.

Reply:
My envisioned lifecycle for this is:

v3:
`--strict-single-hook=false` is the default

v4: ???
`--strict-single-hook=true` is the default

v5:
remove the flag

Reply:
I gave it some thought, and I think I agree with you.